crates/swc_html_minifier/src/lib.rs

```diff
@@ -1915,6 +1948,168 @@ impl Minifier<'_> {
+                _ => return None,
+            }
+        };
+
+        // Avoid compress potential invalid JS
+        if !left_errors.is_empty() {
+            return None;
+        }
+
+        let left_unresolved_mark = Mark::new();
+        let left_top_level_mark = Mark::new();
+
+        swc_ecma_visit::VisitMutWith::visit_mut_with(
+            &mut left_program,
+            &mut swc_ecma_transforms_base::resolver(
+                left_unresolved_mark,
+                left_top_level_mark,
+                false,
+            ),
+        );
+
+        // Right
+        let mut right_errors: Vec<_> = vec![];
+        let right_fm = cm.new_source_file(FileName::Anon, right);
+
+        let mut right_program = if is_modules {
+            match swc_ecma_parser::parse_file_as_module(
+                &right_fm,
+                syntax,
+                target,
+                Some(&comments),
+                &mut right_errors,
+            ) {
+                Ok(module) => swc_ecma_ast::Program::Module(module),
+                _ => return None,
+            }
+        } else {
+            match swc_ecma_parser::parse_file_as_script(
+                &right_fm,
+                syntax,
+                target,
+                Some(&comments),
+                &mut right_errors,
+            ) {
+                Ok(script) => swc_ecma_ast::Program::Script(script),
+                _ => return None,
+            }
+        };
+
+        // Avoid compress potential invalid JS
+        if !right_errors.is_empty() {
+            return None;
+        }
+
+        let right_unresolved_mark = Mark::new();
+        let right_top_level_mark = Mark::new();
+
+        swc_ecma_visit::VisitMutWith::visit_mut_with(
+            &mut right_program,
+            &mut swc_ecma_transforms_base::resolver(
+                right_unresolved_mark,
+                right_top_level_mark,
+                false,
+            ),
+        );
+
+        // Merge
+        match &mut left_program {
+            swc_ecma_ast::Program::Module(left_program) => match right_program {
+                swc_ecma_ast::Program::Module(right_program) => {
+                    left_program.body.extend(right_program.body);
+                }
+                _ => {
+                    unreachable!();
+                }
+            },
+            swc_ecma_ast::Program::Script(left_program) => match right_program {
+                swc_ecma_ast::Program::Script(right_program) => {
+                    left_program.body.extend(right_program.body);
+                }
+                _ => {
+                    unreachable!();
+                }
+            },
+        }
+
+        if is_modules {
+            swc_ecma_visit::VisitMutWith::visit_mut_with(
+                &mut left_program,
+                &mut swc_ecma_transforms_base::hygiene::hygiene(),
+            );
+        }
+
+        let left_program = swc_ecma_visit::FoldWith::fold_with(
```

Comment:
Please use `left_program.fold_with(&mut ...)` instead

Reply:
I don't use this because I will need to use `use swc_ecma_visit::FoldWith;`, but there are ecma/css/html crates here, so I tried to avoid renaming using `as` and use it directly